src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -275,10 +276,13 @@ export class ConnectionDialogService implements IConnectionDialogService {
 			} else if (connectionResult && connectionResult.errorHandled) {
 				this._connectionDialog.resetConnection();
 				this._logService.debug(`ConnectionDialogService: Error handled and connection reset - Error: ${connectionResult.errorMessage}`);
-			} else {
+			} else if (connectionResult.errorCode !== Constants.sqlPasswordErrorCode) {
```

Comment:
We should only do this for MSSQL providers - if another provider happened to give an error code with the same number that isn't necessarily going to mean that a password change is required (not to mention currently there doesn't seem to be any sort of generic way for providers to actually hook into this function as I discussed before. It's specific to just MSSQL)